include/range/v3/view/interface.hpp

```diff
@@ -161,16 +161,16 @@ namespace ranges
         }
         /// If the size of the range is known at compile-time and finite,
         /// return it.
-        template<bool True = true>
+        template<bool True = true, int = 42>
```

Comment:
Por què?

Reply:
[[over.load]/2.2](http://eel.is/c++draft/over.load#2.2):

> Likewise, member function template declarations with the same name, the same parameter-type-list, and the same template parameter lists cannot be overloaded if any of them is a static member function template declaration.

See? MSVC is *more* conforming than clang and GCC. (Despite that the IS should be fixed to agree with what clang and GCC do here, and to allow for differing constraints)

Reply:
Wow. That's a real thing? 😕

Reply:
Yeah. We used to reject *any* static + non-static overloads - regardless of template-ness. I tried to convince the compiler guys to implement more relaxed rules that would allow overloads with differing constraints, but they chose to instead implement the *exact* rule as specified in [over.load]/2.2.